packages/starknet-snap/src/rpcs/executeTxn.ts

```diff
@@ -0,0 +1,397 @@
+          value: this.network.name,
+          markdown: false,
+        }),
+      ),
+    );
+
+    // Divider before calls information
+    components.push(divider());
+
+    // Iterate over each call in the calls array
+    for (const call of calls) {
+      const { contractAddress, calldata, entrypoint } = call;
+      const contractFuncName = entrypoint;
+
+      // Display contract address
+      components.push(
+        row(
+          'Contract',
+          text({
+            value: contractAddress,
+            markdown: false,
+          }),
+        ),
+      );
+
+      // Display call data
+      components.push(
+        row(
+          'Call Data',
+          text({
+            value: JSON.stringify(calldata, null, 2),
+            markdown: false,
+          }),
+        ),
+      );
+
+      // If the contract is an ERC20 token and the function is 'transfer', display sender, recipient, and amount
+      const tokenStateManager = new TokenStateManager();
+
+      const token = await tokenStateManager.getToken({
+        address: contractAddress,
+        chainId: this.network.chainId,
+      });
+      if (token && contractFuncName === 'transfer' && calldata) {
+        try {
+          const senderAddress = address;
+          const recipientAddress = calldata[0]; // Assuming the first element in calldata is the recipient
+          let amount = '';
+
+          if ([3, 6, 9, 12, 15, 18].includes(token.decimals)) {
+            amount = convert(calldata[1], -1 * token.decimals, 'ether');
+          } else {
+            amount = (
+              Number(calldata[1]) * Math.pow(10, -1 * token.decimals)
+            ).toFixed(token.decimals);
+          }
+
+          components.push(
+            row(
+              'Sender Address',
+              text({
+                value: senderAddress,
+                markdown: false,
+              }),
+            ),
+            row(
+              'Recipient Address',
+              text({
+                value: recipientAddress,
+                markdown: false,
+              }),
+            ),
+            row(
+              'Amount',
+              text({
+                value: amount,
+                markdown: false,
+              }),
+            ),
+          );
+        } catch (error) {
+          console.error('Error processing ERC20 transfer:', error);
+        }
+      }
+
+      // Divider after each call
+      components.push(divider());
+    }
+
+    // Display details such as Nonce and Version if available
+    if (details?.nonce) {
+      components.push(
+        row(
+          'Nonce',
+          text({
+            value: details.nonce.toString(),
+            markdown: false,
+          }),
+        ),
+      );
+    }
+    if (details?.version) {
+      components.push(
+        row(
+          'Version',
+          text({
+            value: details.version.toString(),
+            markdown: false,
+          }),
+        ),
+      );
+    }
```

Comment:
```suggestion
```
the end user is not concern the version and nonce, so we dont need to display it